shell/pages/c/_cluster/uiplugins/index.vue

```diff
@@ -449,6 +455,27 @@ export default {
       const shouldHaveLoaded = (installed || []).filter((p) => !this.uiErrors[p.name] && !p.builtin);
       let changes = 0;
 
+      // An install occured and a plugin was added, let add all annotations to the UIPlugin resource (useful for other checks)
+      if (neu?.length === 1 && !installed.find((x) => neu[0].id === x.id) &&
```

Comment:
> * `neu?.length === 1` - this only works if there's a single, non-built in plugin. if there's more than one it will never fire?
> * what happens with `latestInstalledPlugin` if two are installed at the same time?
> * from what i can tell it's persisting (plugin.save) package annotations in the ui-plugin metadata. this doesn't feel right. shouldn't the plugin's package information be available at other points?

Replying to your comments:
1) you are right. I don't know how I missed this. Will adjust accordingly.

2) First point: we cannot install two at the same time. The logic that copies the annotations are tied to a watcher, which will fire every time a plugin is installed, so it only runs once per install. Once it runs on this watcher, it's reset to `undefined`. It's again populated once an install occurs again. So, it should all be covered and work as expected.

3) It's the only way we can persist the data as it's the only resource available to us that is tied to plugins. Afaik, there's no other alternative and it's only metadata outside of the scope of the actual `uiplugins.metadata` field. This is something that is going to be useful to us to do more checks in the future.

Reply:
we need to run through this together. also spotted if the user navigates away from the page the extra information is never applied